cmd/solvercli/cmd/balances.go

```diff
@@ -0,0 +1,331 @@
+	balancesCmd.Flags().String("osmosis-address", "", "Optional Osmosis address to check balances for instead of config address")
+}
+
+func getBalances(ctx context.Context,
+	usdcBalances map[string]*ChainBalance,
+	gasBalances map[string]*ChainGasBalance,
+	customBalances map[string][]*ChainBalance,
+	totalUSDCBalance *big.Int,
+	totalCustomAssetsUSDValue *big.Float,
+	cmd *cobra.Command) error {
+	request, err := buildBalancesRequest(ctx, cmd)
+	if err != nil {
+		return fmt.Errorf("building balances request: %w", err)
+	}
+
+	skipResp, err := fetchBalances(ctx, request)
+	if err != nil {
+		return fmt.Errorf("fetching balances: %w", err)
+	}
+
+	lmt.Logger(ctx).Info("resp", zap.Any("", skipResp))
+	for chainID, chainResp := range skipResp.Chains {
+		chainConfig, err := config.GetConfigReader(ctx).GetChainConfig(chainID)
+		if err != nil {
+			return fmt.Errorf("getting chain config for %s: %w", chainID, err)
+		}
+
+		if err := processUSDCBalance(chainID, chainConfig, chainResp, usdcBalances, totalUSDCBalance); err != nil {
+			return fmt.Errorf("processing USDC balance: %w", err)
+		}
+
+		if err := processGasBalance(ctx, chainID, chainConfig, chainResp, gasBalances); err != nil {
+			return fmt.Errorf("processing gas balance: %w", err)
+		}
+
+		if err := processCustomBalances(chainID, chainConfig, chainResp, customBalances, totalCustomAssetsUSDValue); err != nil {
+			return fmt.Errorf("processing custom balances: %w", err)
+		}
+	}
+
+	return nil
+}
+
+func buildBalancesRequest(ctx context.Context, cmd *cobra.Command) (*skipgo.BalancesRequest, error) {
+	request := &skipgo.BalancesRequest{
+		Chains: make(map[string]skipgo.ChainRequest),
+	}
+
+	customAssetMap := make(map[string][]string)
+	if customAssetsFlag := cmd.Flags().Lookup("custom-assets").Value.String(); customAssetsFlag != "" {
+		if err := json.Unmarshal([]byte(customAssetsFlag), &customAssetMap); err != nil {
+			return nil, fmt.Errorf("parsing custom-assets JSON: %w", err)
+		}
+	}
+
+	evmAddress := cmd.Flags().Lookup("evm-address").Value.String()
+	osmosisAddress := cmd.Flags().Lookup("osmosis-address").Value.String()
+
+	chains := config.GetConfigReader(ctx).Config().Chains
+	for _, chain := range chains {
+		chainConfig, err := config.GetConfigReader(ctx).GetChainConfig(chain.ChainID)
+		if err != nil {
+			return nil, fmt.Errorf("getting chain config for %s: %w", chain.ChainID, err)
+		}
+
+		address := chainConfig.SolverAddress
+		if chainConfig.Type == config.ChainType_EVM && evmAddress != "" {
+			address = evmAddress
+		} else if chainConfig.Type == config.ChainType_COSMOS && osmosisAddress != "" {
+			address = osmosisAddress
+		}
+
+		var gasTokenDenom string
+		if chainConfig.Type == config.ChainType_COSMOS {
+			gasTokenDenom = chainConfig.Cosmos.GasDenom
+		} else if chainConfig.Type == config.ChainType_EVM {
+			gasTokenDenom = chainConfig.ChainName + "-native"
```

Comment:
this is a janky solution, but our balances endpoint doesn't accept ETH as a denom, and returns a balance but no decimals/value_usd for a 0x000.. address. This wasn't an issue previously because we were using rpc to get the gas balance.

Two alternatives are: updating our balances endpoint to accept eth parameter or handle the 0x000 address more gracefully (by setting the usd_value and decimals field in return), or we could alternatively add an eth_representation field in our config but I didnt want to add more noise in our config since there's alot there already @dhfang @mattac21

Reply:
did you end up getting this sorted by passing `ethereum-native` as the denom? or `arbitrum-native`, `optimism-native`, etc?

Reply:
arbitrum-native, etc.

ChainName + "-native"

Reply:
oh duh i see what you're saying now, tbh i think its fine as is. maybe the logic could be moved out to a function on its own, but its probably fine